doc: changelog and migration guide for 1.14.0 release (#1768)
mongodb/terraform-provider-mongodbatlas: CHANGELOG.md

# Changelog

> ## [v1.14.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.14.0) (2023-12-18)

Reviewer: ```suggestion
## [v1.14.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.14.0) (2023-12-19)
```

[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.13.1...v1.14.0)

**Breaking changes:**

> - fix!: replaces .String\(\) method with internal method .TimeToString\(\) to align formating with Atlas API. [\#1699](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1699) ([marcosuma](https://github.com/marcosuma))

Reviewer: ```suggestion
- fix!: replaces .String\(\) method with internal method .TimeToString\(\) to align formating with Atlas API. [\#1699](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1699) ([marcosuma](https://github.com/marcosuma))
   - Fixes date fields that were not compliant to the ISO 8601 timestamp format in UTC, in line with the documentation.
```

> - feat!: new required attributes `authorized_user_first_name` and `authorized_user_last_name` in `mongodbatlas_backup_compliance_policy` resource. [\#1655](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1655) ([marcosuma](https://github.com/marcosuma))

Reviewer: ```suggestion
- feat!: new required attributes `authorized_user_first_name` and `authorized_user_last_name` in `mongodbatlas_backup_compliance_policy` resource. [\#1655](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1655) ([marcosuma](https://github.com/marcosuma))
   - Adds first and last name in `mongodbatlas_backup_compliance_policy` resource
```

**Enhancements**

- feat: new `mongodbatlas_stream_instance` resource [\#1685](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1685) ([AgustinBettati](https://github.com/AgustinBettati))
- feat: new `mongodbatlas_stream_instance` and `mongodbatlas_stream_instances` data sources [\#1689](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1689) [\#1701](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1701) ([AgustinBettati](https://github.com/AgustinBettati))
- feat: new `mongodbatlas_stream_connection` resource [\#1736](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1736) ([AgustinBettati](https://github.com/AgustinBettati))
- feat: new `mongodbatlas_stream_connection` and `mongodbatlas_stream_connections` data sources [\#1757](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1757) ([AgustinBettati](https://github.com/AgustinBettati))

**Bug Fixes**

> - fix: Doesn't disable X.509 in the project when mongodbatlas\_x509\_authentication\_database\_user resource is deleted [\#1760](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1760) ([lantoli](https://github.com/lantoli))

Reviewer: ```suggestion
- fix: Doesn't disable X.509 in the project when `mongodbatlas_x509_authentication_database_user` resource is deleted [\#1760](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1760) ([lantoli](https://github.com/lantoli))
```

- fix: adjust root `project_id` attribute in `mongdbatlas_project_api_key` resource to optional [\#1664](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1664) ([AgustinBettati](https://github.com/AgustinBettati))
- fix: define `project_assignment` block in `mongodbatlas_project_api_key` as required to avoid plugin crash [\#1663](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1663) ([AgustinBettati](https://github.com/AgustinBettati))
- fix: Fixes cluster update when adding replication specs [\#1755](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1755) ([lantoli](https://github.com/lantoli))
- fix: Provider crash when an invalid role name is specified to `mongodbatlas_project_api_key` [\#1720](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1720) ([andreaangiolillo](https://github.com/andreaangiolillo))
- fix: project delete function uses correct string representation of id [\#1733](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1733) ([oarbusi](https://github.com/oarbusi))

> **Bug Fixes for Terraform Version 1.0.8**

Reviewer: ```suggestion
**Bug Fixes for HashiCorp Terraform Version 1.0.8**
```

> - fix: Fixes some tests in search\_index test group in older TF versions [\#1758](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1758) ([lantoli](https://github.com/lantoli))

Reviewer: Believe we are a missing few: 

serverless
cloud_provider_access
org_invitation
project_api_key
third_party_integration

more details in: https://mongodb.slack.com/archives/C04K3F1EGGK/p1702653073328139?thread_ts=1702651967.537649&cid=C04K3F1EGGK

Reviewer 2: that's correct, sorry the PR titles don't cover all of them as they talk about test groups not resources.

Author: perfect, defined resource level details here

- fix: Fixes config test group in older TF versions [\#1748](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1748) ([lantoli](https://github.com/lantoli))
- fix: Fixes network test group in older TF versions [\#1741](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1741) ([lantoli](https://github.com/lantoli))
- fix: Fixes serverless test group in older TF versions [\#1740](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1740) ([lantoli](https://github.com/lantoli))

**Deprecations and Removals**

- deprecate: optional root `project_id` attribute in `mongdbatlas_project_api_key` is deprecated [\#1665](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1665) ([AgustinBettati](https://github.com/AgustinBettati))

**Closed Issues**

> - mongodbatlas\_alert\_configuration failing to be created on apply due to METRIC\_TYPE\_UNSUPPORTED for DISK\_PARTITION\_ alerts 400 error [\#1716](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1716)

Reviewer 3: ```suggestion
- `mongodbatlas_alert_configuration` failing to be created on apply due to `METRIC_TYPE_UNSUPPORTED` for `DISK_PARTITION_` alerts 400 error [\#1716](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1716)
```

- Provider produced inconsistent result after apply [\#1707](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1707)

> -  Error: Provider produced inconsistent final plan for mongodbatlas\_privatelink\_endpoint\_service [\#1690](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1690)

Reviewer 3: ```suggestion
-  Error: Provider produced inconsistent final plan for `mongodbatlas_privatelink_endpoint_service` [\#1690](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1690)
```

- Provider not working with Secret Manager [\#1683](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1683)
- In the `mongodbatlas_advanced_cluster` ressource, forbidden characters for the  values of the tags are allowed by the provider and fail on apply. [\#1668](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1668)

**Internal Improvements**

> - test: unit test for project\_ip\_access\_list resource [\#1756](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1756) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- test: unit test for `project_ip_access_list` resource [\#1756](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1756) ([oarbusi](https://github.com/oarbusi))
```

- chore: CLOUDP-215162: Update Jira GitHub Action to update Ticket status based on the issue [\#1754](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1754) ([andreaangiolillo](https://github.com/andreaangiolillo))

> - chore: migrate project\_ip\_access\_list to new SDK [\#1753](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1753) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- chore: migrate `project_ip_access_list` to new SDK [\#1753](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1753) ([oarbusi](https://github.com/oarbusi))
```

- chore: fix examples job by defining beta flag [\#1751](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1751) ([AgustinBettati](https://github.com/AgustinBettati))

> - test: unit test encryption\_at\_rest resource [\#1750](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1750) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- test: unit test `encryption_at_rest` resource [\#1750](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1750) ([oarbusi](https://github.com/oarbusi))
```

- chore: Bump tj-actions/verify-changed-files from 7b7a3b8db9077729f56bd82fced85f4b0ee67bcd to 5eff60fda839b96c3e34d8239dffb116c900582c [\#1747](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1747) ([dependabot[bot]](https://github.com/apps/dependabot))
- chore: Bump actions/stale from 8 to 9 [\#1746](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1746) ([dependabot[bot]](https://github.com/apps/dependabot))
- chore: Bump actions/setup-go from 4 to 5 [\#1745](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1745) ([dependabot[bot]](https://github.com/apps/dependabot))
- chore: Bump github.com/aws/aws-sdk-go from 1.48.13 to 1.49.1 [\#1744](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1744) ([dependabot[bot]](https://github.com/apps/dependabot))
- chore: new scaffold command for creating resources/data sources [\#1739](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1739) ([AgustinBettati](https://github.com/AgustinBettati))

> - test: add unit test to database\_user resource [\#1738](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1738) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- test: add unit test to `database_user` resource [\#1738](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1738) ([oarbusi](https://github.com/oarbusi))
```

- chore: lint error [\#1734](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1734) ([andreaangiolillo](https://github.com/andreaangiolillo))

> - doc: CLOUDP-215923: \[TF\] Remove sunset resources from documentation [\#1732](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1732) ([andreaangiolillo](https://github.com/andreaangiolillo))

Reviewer 3: ```suggestion
- doc: CLOUDP-215923: Removes sunset resources from documentation [\#1732](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1732) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

- chore: define atlas stream functionality under beta flag environment variable [\#1726](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1726) ([AgustinBettati](https://github.com/AgustinBettati))

> - chore: migrate encryption\_at\_rest resource to new SDK [\#1725](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1725) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- chore: migrates encryption_at_rest resource to new SDK [\#1725](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1725) ([oarbusi](https://github.com/oarbusi))
```

- chore: Bump github.com/aws/aws-sdk-go from 1.48.2 to 1.48.7 [\#1678](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1678) ([dependabot[bot]](https://github.com/apps/dependabot))
- chore: Bump github.com/spf13/cast from 1.5.1 to 1.6.0 [\#1677](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1677) ([dependabot[bot]](https://github.com/apps/dependabot))
- refactor: adjust project\_api\_key to new file structure [\#1676](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1676) ([AgustinBettati](https://github.com/AgustinBettati))
- chore: remove references of legacy mongodbatlas package in make file [\#1675](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1675) ([AgustinBettati](https://github.com/AgustinBettati))
- chore: Updates Atlas Go SDK [\#1674](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1674) ([github-actions[bot]](https://github.com/apps/github-actions))
- chore: Increases project list limit for clean-up Github action [\#1673](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1673) ([lantoli](https://github.com/lantoli))

> - test: add unit test to alert\_configuration [\#1670](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1670) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- test: Adds unit test to `alert_configuration` [\#1670](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1670) ([oarbusi](https://github.com/oarbusi))
```

- feat: Restructures files [\#1657](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1657) ([lantoli](https://github.com/lantoli))
- feat: Runs Acceptance and Migration tests for different Terraform CLI versions [\#1688](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1688) ([lantoli](https://github.com/lantoli))
- feat: Allows to choose Terraform version for Acceptance and Migration Tests, default to latest version [\#1684](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1684) ([lantoli](https://github.com/lantoli))
- fix: Fixes config flaky tests [\#1680](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1680) ([lantoli](https://github.com/lantoli))
- fix: Fixes project flaky tests [\#1669](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1669) ([lantoli](https://github.com/lantoli))
- fix: Runs only test group if specified in Github Action acceptance or migration tests [\#1730](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1730) ([lantoli](https://github.com/lantoli))

> - fix: Fixes migration tests for backup\_online\_archive [\#1724](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1724) ([lantoli](https://github.com/lantoli))

Reviewer 3: ```suggestion
- fix: Fixes migration tests for `backup_online_archive` [\#1724](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1724) ([lantoli](https://github.com/lantoli))
```

# Changelog

## [v1.14.0](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.14.0) (2023-12-19)

[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.13.1...v1.14.0)

**Breaking changes:**

> - fix!: replaces .String\(\) method with internal method .TimeToString\(\) to align formating with Atlas API. [\#1699](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1699) ([marcosuma](https://github.com/marcosuma))

Reviewer 4: ```suggestion
- fix!: replaces .String\(\) method with internal method .TimeToString\(\) to align formatting with the Atlas API. [\#1699](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1699) ([marcosuma](https://github.com/marcosuma))
```

>    - Fixes date fields that were not compliant to the ISO 8601 timestamp format in UTC, in line with the documentation.

Reviewer 2: does this line have leading blanks?

Author: Yes this is intentional as it is a nested bullet to provide more detail on the breaking change

**Enhancements**

- feat: new `mongodbatlas_stream_instance` resource [\#1685](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1685) ([AgustinBettati](https://github.com/AgustinBettati))
- feat: new `mongodbatlas_stream_instance` and `mongodbatlas_stream_instances` data sources [\#1689](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1689) [\#1701](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1701) ([AgustinBettati](https://github.com/AgustinBettati))
- feat: new `mongodbatlas_stream_connection` resource [\#1736](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1736) ([AgustinBettati](https://github.com/AgustinBettati))
- feat: new `mongodbatlas_stream_connection` and `mongodbatlas_stream_connections` data sources [\#1757](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1757) ([AgustinBettati](https://github.com/AgustinBettati))

**Bug Fixes**

- fix: Doesn't disable X.509 in the project when `mongodbatlas_x509_authentication_database_user` resource is deleted [\#1760](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1760) ([lantoli](https://github.com/lantoli))

> - fix: adjust root `project_id` attribute in `mongdbatlas_project_api_key` resource to optional [\#1664](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1664) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 3: ```suggestion
- fix: Converts root `project_id` attribute to optional in `mongdbatlas_project_api_key` resource [\#1664](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1664) ([AgustinBettati](https://github.com/AgustinBettati))
```

> - fix: define `project_assignment` block in `mongodbatlas_project_api_key` as required to avoid plugin crash [\#1663](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1663) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 3: ```suggestion
- fix: Defines `project_assignment` block in `mongodbatlas_project_api_key` as required to avoid plugin crash [\#1663](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1663) ([AgustinBettati](https://github.com/AgustinBettati))
```

> - fix: define `project_assignment` block in `mongodbatlas_project_api_key` as required to avoid plugin crash [\#1663](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1663) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 3: ```suggestion
- fix: Defines `project_assignment` block as required in `mongodbatlas_project_api_key`  to avoid plugin crash [\#1663](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1663) ([AgustinBettati](https://github.com/AgustinBettati))
```

- fix: Fixes cluster update when adding replication specs [\#1755](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1755) ([lantoli](https://github.com/lantoli))

> - fix: Provider crash when an invalid role name is specified to `mongodbatlas_project_api_key` [\#1720](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1720) ([andreaangiolillo](https://github.com/andreaangiolillo))

Reviewer 3: ```suggestion
- fix: Provider crashes when an invalid role name is specified to `mongodbatlas_project_api_key` [\#1720](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1720) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

> - fix: project delete function uses correct string representation of id [\#1733](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1733) ([oarbusi](https://github.com/oarbusi))

Reviewer 3: ```suggestion
- fix: Fixes string representation of id for project delete function [\#1733](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1733) ([oarbusi](https://github.com/oarbusi))
```

**Bug Fixes for HashiCorp Terraform Version 1.0.8**

> - fix: Fixes `mongodbatlas_cloud_provider_access`, `mongodbatlas_org_invitation`, `mongodbatlas_project_api_key`, `mongodbatlas_third_party_integration` in older TF versions [\#1748](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1748) ([lantoli](https://github.com/lantoli))

Reviewer 4: ```suggestion
- fix: Fixes `mongodbatlas_cloud_provider_access`, `mongodbatlas_org_invitation`, `mongodbatlas_project_api_key`, `mongodbatlas_third_party_integration` in older Terraform versions [\#1748](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1748) ([lantoli](https://github.com/lantoli))
```

> - fix: Fixes `mongodbatlas_network_container`, `mongodbatlas_private_endpoint_regional_mode` in older TF versions [\#1741](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1741) ([lantoli](https://github.com/lantoli))

Reviewer 4: ```suggestion
- fix: Fixes `mongodbatlas_network_container`, `mongodbatlas_private_endpoint_regional_mode` in older Terraform versions [\#1741](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1741) ([lantoli](https://github.com/lantoli))
```

> - fix: Fixes `mongodbatlas_serverless_instance` in older TF versions [\#1740](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1740) ([lantoli](https://github.com/lantoli))

Reviewer 4: ```suggestion
- fix: Fixes `mongodbatlas_serverless_instance` in older Terraform versions [\#1740](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1740) ([lantoli](https://github.com/lantoli))
```

> - fix: Fixes some tests in search_index test group in older TF versions [\#1758](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1758) ([lantoli](https://github.com/lantoli))

Reviewer 3: ```suggestion
- fix: Fixes some tests in `search_index` test group in older TF versions [\#1758](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1758) ([lantoli](https://github.com/lantoli))
```

> - fix: Fixes some tests in search_index test group in older TF versions [\#1758](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1758) ([lantoli](https://github.com/lantoli))

Reviewer 4: ```suggestion
- fix: Fixes some tests in search_index test group in older Terraform versions [\#1758](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1758) ([lantoli](https://github.com/lantoli))
```

**Deprecations and Removals**

> - deprecate: optional root `project_id` attribute in `mongdbatlas_project_api_key` is deprecated [\#1665](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1665) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 3: ```suggestion
- deprecate: Deprecates optional root `project_id` attribute in `mongdbatlas_project_api_key` [\#1665](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1665) ([AgustinBettati](https://github.com/AgustinBettati))
```

- doc: CLOUDP-215923: Removes sunset resources from documentation [\#1732](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1732) ([andreaangiolillo](https://github.com/andreaangiolillo))
- chore: define atlas stream functionality under beta flag environment variable [\#1726](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1726) ([AgustinBettati](https://github.com/AgustinBettati))
- chore: `migrate encryption_at_rest` resource to new SDK [\#1725](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1725) ([oarbusi](https://github.com/oarbusi))
- chore: migrate database user resource to new SDK [\#1723](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1723) ([oarbusi](https://github.com/oarbusi))
- doc: CLOUDP-216288 - Update the warning message in the cluster resource to mention a bug on the container ids [\#1719](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1719) ([andreaangiolillo](https://github.com/andreaangiolillo))
- chore: Uses official CLI GH action in TF clean-up workflow [\#1718](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1718) ([lantoli](https://github.com/lantoli))

> - chore: Bump tj-actions/verify-changed-files from 1e75cac4ffa7ea5879addde1869f8fca09fce4c1 to 7b7a3b8db9077729f56bd82fced85f4b0ee67bcd [\#1713](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1713) ([dependabot[bot]](https://github.com/apps/dependabot))

Reviewer 2: knitpit: I would put all bumps together at the end of the section

Author: adjusted

- chore: Updates Atlas Go SDK [\#1674](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1674) ([github-actions[bot]](https://github.com/apps/github-actions))
- chore: Increases project list limit for clean-up Github action [\#1673](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1673) ([lantoli](https://github.com/lantoli))
- test: Adds unit test to `alert_configuration` [\#1670](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1670) ([oarbusi](https://github.com/oarbusi))
- chore: adjust CI change detection file paths after file restructure [\#1667](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1667) ([AgustinBettati](https://github.com/AgustinBettati))
- doc: Updates contributing file with code and test best practices [\#1666](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1666) ([lantoli](https://github.com/lantoli))
- chore: Adds a website make goal to preview doc changes [\#1662](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1662) ([lantoli](https://github.com/lantoli))

> - test: including unit testing for search deployments state transition logic and model conversions [\#1653](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1653) ([AgustinBettati](https://github.com/AgustinBettati))

Reviewer 2: ```suggestion
- test: Includes unit testing for search deployments state transition logic and model conversions [\#1653](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1653) ([AgustinBettati](https://github.com/AgustinBettati))
```
